### PR TITLE
[11.0][IMP] partner_sector: Better handling of the transition for this module

### PR DESCRIPTION
**Pull request by pedrobaeza**
* Rename tables instead of copying data
* Remove the need to reassign many2one references.

@Tecnativa

### CODE REVIEW

File: odoo/addons/base/migrations/11.0.1.3/pre-migration.py [hunk 1]
```diff
@@ -25,6 +25,22 @@
...
+    if openupgrade.table_exists(env.cr, 'res_partner_sector'):
+        # Module `partner_sector` was installed in 10.0
+        openupgrade.rename_models(
+            env.cr, [('res.partner.sector', 'res.partner.industry')],
+        )
+        openupgrade.rename_tables(
+            env.cr, [(('res_partner_sector', 'res_partner_industry'))]
+        )
+        openupgrade.rename_fields(
+            env.cr, [
```
Comment (MiquelRForgeFlow) [inline odoo/addons/base/migrations/11.0.1.3/pre-migration.py]: should be `env` instead of `env.cr`

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]
LGTM

File: odoo/addons/base/migrations/11.0.1.3/pre-migration.py (hunk 1 above)
Comment (pedrobaeza) [reply odoo/addons/base/migrations/11.0.1.3/pre-migration.py]: Totally! fixed

Comment (MiquelRForgeFlow) [reply odoo/addons/base/migrations/11.0.1.3/pre-migration.py]: but the problem I see is that in the base pre-migration, the `env` still doesn't have the base models if remember correctly. 🤔

Comment (pedrobaeza) [reply odoo/addons/base/migrations/11.0.1.3/pre-migration.py]: Uhm, good question... let me check if we need it for something specific.

Comment (pedrobaeza) [reply odoo/addons/base/migrations/11.0.1.3/pre-migration.py]: There's no problem:

https://github.com/OCA/openupgradelib/blob/12b1ed9c7d2c354499ba5f89067f908f0f8f94ad/openupgradelib/openupgrade.py#L413-L415

Comment (MiquelRForgeFlow) [reply odoo/addons/base/migrations/11.0.1.3/pre-migration.py]: good

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**